timesketch/lib/analyzers/interface.py

```diff
@@ -1150,3 +1165,196 @@ def get_kwargs(cls):
     def run(self):
         """Entry point for the analyzer."""
         raise NotImplementedError
+
+
+class AnalyzerOutputException(Exception):
+    """Analyzer output exception."""
+
+
+class AnalyzerOutput:
+    """A class to record timesketch analyzer output.
+
+    Attributes:
+        platform (str): [Required] Analyzer platfrom.
+        analyzer_identifier (str): [Required] Unique analyzer identifier.
+        analyzer_name (str): [Required] Analyzer display name.
+        result_status (str): [Required] Analyzer result status.
+            Valid values are success, no-findings, error.
+        dfiq_question_id (str): [Optional] Digital Forensics Investigative
+            Question (DFIQ) questions ID.
+        dfiq_question_conclusion (str): [Optional] DFQI questions answer/
+            conclusion.
+        result_priority (str): [Required] Priority of the result based on the
+            analysis findings. Valid values are NOTE (default), LOW, MEDIUM, HIGH.
+        result_summary (str): [Required] A summary statement of the analyzer
+            finding. A result summary must exist even if there is no finding.
+        result_markdown (str): [Optional] A detailed information about the
+            analyzer finding in a markdown format.
+        references (List[str]): [Optional] A list of references about the
+            analyzer or the issue the analyzer attempts to address.
+        attributes (dict): [Optional] A list of object that holds the
```

Comment:
Is this a list or a dict? The docstring is ambiguous

Reply:
And what type of 
```suggestion
        attributes (dict): [Optional] A list of objects that holds the
```

Reply:
and what type of objects is this?

Reply:
good catch, thanks. It is supposed to be a dictionary of key value pairs.
```suggestion
        result_attributes (dict): [Optional] A dict of key : value pairs that holds additional
```